### PR TITLE
Create a build configuraition to build OOB logging site extension package

### PR DESCRIPTION
**Pull request by pakrym**
Initial work to implement https://github.com/aspnet/AzureIntegration/issues/227

### CODE REVIEW

File: .azure/pipelines/site-extensions.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    # This variable is required by MicroBuildSigningPlugin to determine permissions for codesigning.
+    TeamName: AspNetCore
+
+    # SignType = { test, real }
+    # This is prefixed underscore because variables automatically become environment variables (and therefore MSBuild properties),
+    # and this one was causing issues in MSBuild projects which use the $(SignType) MSbuild prop.
+    _SignType: test
+
+  steps:
+  # - task: MicroBuildSigningPlugin@1
```
Comment (natemcmaster) [inline .azure/pipelines/site-extensions.yml]: You'll need to uncomment this for official builds. You'll also need this task at the end of the build:
```yml
  - task: MicroBuildCleanup@1
    displayName: Cleanup MicroBuild tasks
    condition: always()
```

File: src/SiteExtensions/dependencies.props [hunk 2]
```diff
@@ -0,0 +1,11 @@
+﻿<Project>
+  <PropertyGroup>
+    <MSBuildAllProjects>$(MSBuildAllProjects);$(MSBuildThisFileFullPath)</MSBuildAllProjects>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <MicrosoftAspNetCoreAzureAppServicesSiteExtension21PackageVersion>2.1.1-rtm-31076</MicrosoftAspNetCoreAzureAppServicesSiteExtension21PackageVersion>
+    <MicrosoftAspNetCoreAzureAppServicesSiteExtension22PackageVersion></MicrosoftAspNetCoreAzureAppServicesSiteExtension22PackageVersion>
```
Comment (Eilon) [inline src/SiteExtensions/dependencies.props]: This is empty on purpose?

File: src/SiteExtensions/src/Microsoft.AspNetCore.AzureAppServices.SiteExtension/Microsoft.AspNetCore.AzureAppServices.SiteExtension.csproj [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    <NoPackageAnalysis>true</NoPackageAnalysis>
+    <IncludeBuildOutput>false</IncludeBuildOutput>
+    <IncludeSymbols>false</IncludeSymbols>
+    <IncludeSource>false</IncludeSource>
+    <ContentTargetFolders>content</ContentTargetFolders>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore.AzureAppServices.SiteExtension.2.1" Version="$(MicrosoftAspNetCoreAzureAppServicesSiteExtension21PackageVersion)" PrivateAssets="All" />
+    <!-- <PackageReference Include="Microsoft.AspNetCore.AzureAppServices.SiteExtension.2.2" Version="$(MicrosoftAspNetCoreAzureAppServicesSiteExtension22PackageVersion)" PrivateAssets="All" /> -->
```
Comment (Eilon) [inline src/SiteExtensions/src/Microsoft.AspNetCore.AzureAppServices.SiteExtension/Microsoft.AspNetCore.AzureAppServices.SiteExtension.csproj]: I guess this is related to that empty version above? Is there a plan to un-comment this, or just delete it? Any issue we need to track this?

Comment (pakrym) [reply src/SiteExtensions/src/Microsoft.AspNetCore.AzureAppServices.SiteExtension/Microsoft.AspNetCore.AzureAppServices.SiteExtension.csproj]: I'm waiting for 2.2 branch build to go through and produce this package.

File: src/SiteExtensions/dependencies.props (hunk 2 above)
Comment (pakrym) [reply src/SiteExtensions/dependencies.props]: Waiting for a package to be produced from 2.2 branch.

Comment (Eilon) [reply src/SiteExtensions/dependencies.props]: Got it!

**Comment by pakrym** [comment]
I've added MicroBuildCleanup. It's commented out for now so I can test in the public stamp.

**Comment by pakrym** [comment]
This also consumes both 2.1 and 2.2 packages now.

File: .azure/pipelines/site-extensions.yml (hunk 1 above)
Comment (natemcmaster) [reply .azure/pipelines/site-extensions.yml]: you can also do this to exclude the tasks and other variables in PRs and public builds.

```suggestion
  ${{ if and(ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'PullRequest')) }}:
    - task: MicroBuildSigningPlugin@1
```

Comment (pakrym) [reply .azure/pipelines/site-extensions.yml]: I don't think it's required, it would never run in public after getting merged.

**Comment by pakrym** [comment]
This is ready to get merged and moved to the private project. Ping @natemcmaster

**Review by natemcmaster — APPROVED** [review]

**Pull request merged**